### PR TITLE
use websocket channel for docker driver

### PR DESCRIPTION
**Pull request by rmg**
Connect to strongloop-internal/scrum-nodeops#582

### CODE REVIEW

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
@sam-github this is the minimum change required to get Docker to _support_ WS, but it does not include the changes to make it required and does not include any driver refactoring to reduce duplication of WS related code. Should I continue on this PR with those changes? Or review/merge this and continue on with a follow-up PR?

**Comment by rmg** [comment]
@slnode test please

**Comment by sam-github** [comment]
@rmg no preference, as you wish

File: lib/drivers/docker/index.js [hunk 1]
```diff
@@ -392,3 +419,9 @@ DockerDriver.prototype.getName = function() {
 DockerDriver.prototype.getStatus = function() {
   return 'running';
 };
+
+function publicIp() {
+  var eths = os.networkInterfaces();
+  var addrs = eths.en0 || eths.eth0;
```
Comment (kraman) [inline lib/drivers/docker/index.js]: This runs on the host system right? Seems to be too restrictive. How about search for the first IPv4 addr which is not in the 127.* space.

**Comment by kraman** [comment]
Small nit about the public IP detection. Rest LGTM

File: lib/drivers/docker/index.js (hunk 1 above)
Comment (rmg) [inline lib/drivers/docker/index.js]: This is only meant as a stand-in for something we need to allow the user to configure.. I'll make a note of it.

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
@kraman I rebased with note about the public IP. Does that address your comment well enough to merge?

BTW, b60125f737011f77f5bb8d657430e0c8a4603899 should cover pretty much all the changes you would need to make to central to switch it away from express-ws.

**Comment by kraman** [comment]
Lgtm

**Pull request merged**